### ISSUE TITLE
Is it possible to reorder the input list without resetting all the form data?

### ISSUE THREAD

**Issue by etc-tiago** [Rocketseat/unform#262]
**Description of issue**

I tried to follow the docs of the [complex structure](https://unform.dev/guides/complex-structures) but my goal goes a little further. I have a form with a list of inputs where the values will be transformed into a list for the react-selector, dynamically.

My point of request is in the reordering of items. I'm using `react-sortable-hoc` and when making the move, I get the old and the new index.

I tried to [get the data](https://unform.dev/guides/get-set-field-value) from the form and set only the list data, updated, however, because the input uses defaultValue, it does not update. 

Follow this code, the `react-sortable-hoc` send the params after moved: 
```ts
const sortAction = ({ oldIndex, newIndex }) => {
    if (oldIndex !== newIndex) {
      const form = formRef!.current as any;
      // The required data is `fields[0]`
      const allData = form.getData().fields;
      // get item data
      const item = allData[oldIndex];
      // remove item from list
      const data = [...allData.slice(0, oldIndex), ...allData.slice(oldIndex + 1)];
      // insert item on newIndex position
      const dataWithNewIndex = [...data.slice(0, newIndex), item, ...data.slice(newIndex)];
      // set values in form
      form.setFieldValue('fields', dataWithNewIndex);
    }
  };
```

**Question**
Has any possibility to reorder the input list without reset all form data?

**Comment by diego3g** [comment]
@etc-tiago Can you try to provide a minimal example on Code Sandbox of what you're doing, so i can reproduce it inside my environment? This way i can probably find an way to fix it.

**Comment by etc-tiago** [comment]
@diego3g Thanks for the answer. During the construction of the code sandbox for example, I realized that my code mixed with the context api I was using previously in the formation of the array.

I also realized after looking at the `@unform/core` code that the reason for not updating is that my field may not be registered, as it is a dynamic form:
https://github.com/Rocketseat/unform/blob/57a735b0b8daf8507354ebc81bc453b146e582f2/packages/core/lib/FormProvider.tsx#L37-L46

So to fix the problem I had, it was enough to add an initial element, in case the array was empty.
```tsx
const formData = form.getData()
const fields = formData.fields || [""] // <-this fix the problem
```
```tsx
return (
   <Scope path="fields">
      <SortContainer useDragHandle onSortEnd={sortAction}>
         {fields.map((field: string, index: number) => <SortItem key={index} index={index} props={{ label: field }}/>)}
      </SortContainer>
    </Scope>
)
```
